connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java

```diff
@@ -0,0 +1,298 @@
+        }
+    }
+
+    public void produce(String topic, String value) {
+        produce(topic, null, value);
+    }
+
+    public void produce(String topic, String key, String value) {
+        producer.send(new ProducerRecord<>(topic, key == null ? null : key.getBytes(), value == null ? null : value.getBytes()));
+    }
+
+    public AdminClient createAdminClient() {
+        final Properties adminClientConfig = new Properties();
+        adminClientConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers());
+        final Object listeners = brokerConfig.get(KafkaConfig$.MODULE$.ListenersProp());
+        if (listeners != null && listeners.toString().contains("SSL")) {
+            adminClientConfig.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, brokerConfig.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG));
+            adminClientConfig.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, ((Password) brokerConfig.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG)).value());
+            adminClientConfig.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SSL");
+        }
+        return AdminClient.create(adminClientConfig);
+    }
+
+    /**
+     * Consume at least n records in a given duration or throw an exception.
+     *
+     * @param n the number of expected records in this topic.
+     * @param maxDuration the max duration to wait for these records.
+     * @param topics the topics to subscribe and consume records from.
+     * @return a {@link ConsumerRecords} collection containing at least n records.
+     */
+    public ConsumerRecords<byte[], byte[]> consume(int n, long maxDuration, String... topics) {
+        long num = 1;
+        long duration = maxDuration;
+        // poll in intervals of 250millis.
+        if (duration > 250) {
+            num = 1 + (maxDuration / 250);
+            duration = 250;
+        }
+
+        Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> records = new HashMap<>();
+        int consumedRecords = 0;
+        try (KafkaConsumer<byte[], byte[]> consumer = createConsumer(topics)) {
+            for (int i = 0; i < num; i++) {
+                ConsumerRecords<byte[], byte[]> rec = consumer.poll(Duration.ofMillis(duration));
+                if (rec.isEmpty()) {
+                    continue;
+                }
+                for (TopicPartition partition: rec.partitions()) {
+                    final List<ConsumerRecord<byte[], byte[]>> r = rec.records(partition);
+                    records.computeIfAbsent(partition, t -> new ArrayList<>()).addAll(r);
+                    consumedRecords += r.size();
+                }
+                if (consumedRecords >= n) {
+                    return new ConsumerRecords<>(records);
+                }
+            }
+        }
+
+        throw new RuntimeException("Could not find enough records. found " + consumedRecords + ", expected " + n);
+    }
+
+    public KafkaConsumer<byte[], byte[]> createConsumer(String... topics) {
```

Comment:
It might be better to have a `createConsumer(Map<String, Object> props)` method and `createConsumer()` method that does everything here except the subscribe, and to leave the subscribe call (which is one method invocation) to the caller. (Probably should mention the latter in the JavaDoc for this method.)

This additional control will make it easier for users to create a consumer that fits their needs. And since this method is currently only called from within the `consume(...)` method, there's little downside in having the caller do the subscribe.